SystemInfo/SystemInfoCollectorRegistry.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * File containing the SystemInfoCollectorRegistry class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzSupportToolsBundle\SystemInfo;
+
+use EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector;
+
+/**
+ * A registry of SystemInfoCollectors that uses the SystemInfoCollector's service tag alias to identify the collector.
+ */
+class SystemInfoCollectorRegistry
+{
+    /** @var \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector[] */
+    private $registry = [];
+
+    /**
+     * Adds a SystemInfoCollector with its service tag alias to the registry.
+     *
+     * @param \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector $infoCollector
+     * @param string $alias Service tag alias
+     */
+    public function addToRegistry(SystemInfoCollector $infoCollector, $alias)
+    {
+        $this->registry[$alias] = $infoCollector;
+    }
+
+    /**
+     * Returns the SystemInfoCollector matching the argument.
+     *
+     * @param string $serviceTagAlias A service tag alias to match against. Example: composer
+     *
+     * @return \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector|null The SystemInfoCollector, or null if there is none.
+     */
+    public function getFromRegistry($serviceTagAlias)
+    {
+        if (isset($this->registry[$serviceTagAlias])) {
+            return $this->registry[$serviceTagAlias];
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns all registered SystemInfoCollectors.
+     *
+     * @return \EzSystems\EzSupportToolsBundle\SystemInfo\Collector\SystemInfoCollector[] Hash of SystemInfoCollectors, with service tag alias as keys.
+     */
+    public function getAllFromRegistry()
```

Comment:
To paraphrase you: "A performance improvement I try to apply: instead of adding N method calls, add one method call with N collectors" :P
The CLI command has a usecase for getting all collectors at once. But I'll remove it if you insist :)